src/Auth0Client.ts

```diff
@@ -497,9 +497,13 @@ export default class Auth0Client {
 
       await lock.acquireLock(GET_TOKEN_SILENTLY_LOCK_KEY, 5000);
 
-      const authResult = this.options.useRefreshTokens
-        ? await this._getTokenUsingRefreshToken(getTokenOptions)
-        : await this._getTokenFromIFrame(getTokenOptions);
+      // Only get an access token using a refresh token if:
+      // * refresh tokens are enabled
+      // * no audience has been specified to getTokenSilently (we can only get a token for a new audience when using an iframe)
+      const authResult =
+        this.options.useRefreshTokens && !options.audience
```

Comment:
We have the `aud` claim of the existing token, so we could see if `options.audience` is different (presumably, specifying an audience that is the same as the existing one will work for the refresh grant?)

Reply:
I'm not sure we can check that. We only have the `aud` property from the ID token which will be set to the client ID. The access token may have the correct audience but we can't decode those on the client anyway.

If we could do that check, then yes we could skip this fallback but I can't think of a value to check against that would be reliable enough.